libbeat/common/docker/watcher.go

```diff
@@ -289,17 +296,32 @@ func (w *watcher) watch() {
 				time.Sleep(1 * time.Second)
 				break WATCH
 
+			case <-tickChan.C:
+				if time.Since(time.Unix(w.lastValidTimestamp, 0)) > 10*time.Minute {
+					logp.Info("No events received withing 10 minutes, restarting watch call")
+					w.lastValidTimestamp = time.Now().Unix()
+					time.Sleep(1 * time.Second)
+					break WATCH
+				}
+
 			case <-w.ctx.Done():
 				logp.Debug("docker", "Watcher stopped")
 				w.stopped.Done()
+				tickChan.Stop()
 				return
 			}
 		}
+
+		tickChan.Stop()
 	}
 }
 
 func (w *watcher) listContainers(options types.ContainerListOptions) ([]*Container, error) {
-	containers, err := w.client.ContainerList(w.ctx, options)
+	logp.Debug("docker", "List containers")
+	ctx, cancel := context.WithTimeout(w.ctx, 10*time.Second)
```

Comment:
could you please make all new timeout constants? this should make it easier to fine tune

Reply:
changed